rust/chains/hyperlane-ethereum/src/mailbox.rs

```diff
@@ -199,13 +200,36 @@ where
             metadata.to_vec().into(),
             RawHyperlaneMessage::from(message).to_vec().into(),
         );
-
         let gas_limit = if let Some(gas_limit) = tx_gas_limit {
             gas_limit
         } else {
             tx.estimate_gas().await?.saturating_add(U256::from(100000))
         };
-        Ok(tx.gas(gas_limit))
+        if let Ok((max_fee, max_priority_fee)) = self.provider.estimate_eip1559_fees(None).await {
```

Comment:
This would be a good place to use
```rs
let Ok((max_fee, max_priority_fee)) = self.provider.estimate_eip1559_fees(None).await else {
    return Ok(tx.gas(gas_limit))
};
```